### PR TITLE
snp_defs: add missing fields for `SevFeatures`

### PR DESCRIPTION
**Pull request by 00xc**
Add several missing bits for `SevFeatures`:
* bit 8: VMPL Supervisor Shadow Stack (VmplSSS).
* bit 9: Secure TSC (SecureTscEn).
* bit 14: VMSA Register Protection (VmsaRegProt).
* bit 15: SMT Protection (SmtProtection).

### CODE REVIEW

File: igvm/src/snp_defs.rs [hunk 1]
```diff
@@ -78,8 +83,14 @@ pub struct SevFeatures {
     pub debug_swap: bool,
     pub prevent_host_ibs: bool,
     pub snp_btb_isolation: bool,
-    #[bits(56)]
-    _unused: u64,
+    pub vmpl_supervisor_shadow_stack: bool,
+    pub secure_tsc: bool,
+    #[bits(4)]
+    _unused1: u8,
```
Comment (chris-oo) [inline igvm/src/snp_defs.rs:89]: nit: lets rename these two unused fields to reserved. I guess we didn't do that originally, but that matches better with the spec/common patterns.

**Comment by chris-oo** [comment]
other than the small nit, change looks fine to me! Thanks for your contribution!

**Comment by 00xc** [comment]
@microsoft-github-policy-service agree [company="SUSE"]

**Comment by 00xc** [comment]
@microsoft-github-policy-service agree company="SUSE"

**Comment by 00xc** [comment]
@chris-oo updated with your suggestion

**Comment by chris-oo** [comment]
Thanks! I've gone ahead and enabled automerge. Do you need a release on crates.io with this change?

**Review by chris-oo — APPROVED** [review]

**Comment by 00xc** [comment]
> Thanks! I've gone ahead and enabled automerge. Do you need a release on crates.io with this change?

I think at some point we will want this but it's not urgent, so I'd say not for now.

**Pull request merged**